prow/pjutil/pjutil.go

```diff
@@ -23,36 +23,52 @@ import (
 	"github.com/satori/go.uuid"
 	"github.com/sirupsen/logrus"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
 
 	"k8s.io/test-infra/prow/config"
 	"k8s.io/test-infra/prow/github"
 	"k8s.io/test-infra/prow/kube"
 )
 
 const (
-	jobNameLabel = "prow.k8s.io/job"
-	jobTypeLabel = "prow.k8s.io/type"
-	orgLabel     = "prow.k8s.io/refs.org"
-	repoLabel    = "prow.k8s.io/refs.repo"
-	pullLabel    = "prow.k8s.io/refs.pull"
+	defaultProw = "prow.k8s.io"
+
+	jobNameLabel = "/job"
+	jobTypeLabel = "/type"
+	orgLabel     = "/refs.org"
+	repoLabel    = "/refs.repo"
+	pullLabel    = "/refs.pull"
 )
 
 // NewProwJob initializes a ProwJob out of a ProwJobSpec.
 func NewProwJob(spec kube.ProwJobSpec, labels map[string]string) kube.ProwJob {
+	return NewProwJobWithProwURL(spec, labels, defaultProw)
+}
+
+func NewProwJobWithProwURL(spec kube.ProwJobSpec, labels map[string]string, prow string) kube.ProwJob {
 	allLabels := map[string]string{
-		jobNameLabel: spec.Job,
-		jobTypeLabel: string(spec.Type),
+		prow + jobNameLabel: spec.Job,
+		prow + jobTypeLabel: string(spec.Type),
 	}
 	if spec.Type != kube.PeriodicJob {
-		allLabels[orgLabel] = spec.Refs.Org
-		allLabels[repoLabel] = spec.Refs.Repo
+		allLabels[prow+orgLabel] = spec.Refs.Org
+		allLabels[prow+repoLabel] = spec.Refs.Repo
 		if len(spec.Refs.Pulls) > 0 {
-			allLabels[pullLabel] = strconv.Itoa(spec.Refs.Pulls[0].Number)
+			allLabels[prow+pullLabel] = strconv.Itoa(spec.Refs.Pulls[0].Number)
 		}
 	}
 	for key, value := range labels {
 		allLabels[key] = value
 	}
+
+	// let's validate labels
+	for key, value := range allLabels {
+		if errs := validation.IsValidLabelValue(value); len(errs) > 0 {
+			delete(allLabels, key)
```

Comment:
Is there not something to escape a value to make it a valid label? I don't want to lose UX here

Reply:
IMO we have/or should have presubmit to validate labels in some presubmit to ensure they are valid in the config :-)

Reply:
I made it try to use basename first before delete the key